### ISSUE TITLE
Android: Graphs might not be displayed when using self signed certificate

### ISSUE THREAD

**Issue by tsteur** [piwik/piwik-mobile-2#5330]
A user reported that graphs are not displayed on his Android phone (Nexus 5 running Android 4.4) and I can reproduce this on a Samsung Galaxy S4 running Android 4.4. Need to test it on another Android version to see whether only this version is affected but doubt it. Most likely affects all Android versions. It seems to work fine on iOS.

I'm actually not 100% sure whether the problem is related to the self signed certificate but it must be.  There is no error in the log even when using log level `trace`, and the ImageView component triggers a `load` event which means it loaded the image successful (which it didn't apparently). Instead of the `load` event it should trigger an `error` event but it doesn't => This sounds like a bug in Titanium. I had a look at their issue tracker but couldn't find an issue related to this so far. Need to look again...

Usually we try to load an image by letting the ImageView handle it as it supports caching etc. If any error occurs, eg because of SSL certificate error, we try to load the image manually via XHR with SSL validation disabled (if SSL validation is disabled in settings). As the error event is never triggered it doesn't load the picture using this fallback mode. 

I tried to load the image URL directly via XHR manually and this works. So maybe we should do this always but I'm not sure about possible downsides etc

**Comment by tsteur** [comment]
Should still test for memory issues.

**Comment by tsteur** [comment]
User says he uses SNI... so this might be as well related to SNI

**Comment by tsteur** [comment]
User confirmed it works now.

I did memory checks on iOS and Android and both seems to be ok. The behaviour on Android is interesting. When loading the images manually via a HttpRequest the memory used by the ImageView is freed immediately after the screen containing the image is closed. When letting Android loading the image the memory is not freed immediately but later on demand eg after loading 3 or 4 more graphs (fullscreen) it starts freeing memory (There should be nothing wrong with this behaviour). Maybe we could always use our manual loading behaviour though (currently we do this only if validate SSL is disabled) but want to see whether it works fine or not. Closing issue for now
